app/models/assignment.rb

```diff
@@ -622,4 +622,11 @@ def find_due_dates(type)
     self.due_dates.select {|due_date| due_date.deadline_type_id == DeadlineType.find_by_name(type).id }
   end
 
+  def self.first_submission_reward(assgt_id, team_id)
+    assignment = Assignment.find(assgt_id)
+    if !assignment.nil? && assignment.first_sub_teamid == -1
+      assignment.first_sub_teamid = team_id
```

Comment:
We already have `submission_records` table. We can use that table to check 1st submission or early birds submissions.